x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/forms/index_patterns_form.tsx

```diff
@@ -0,0 +1,158 @@
+    ? {
+        indexLabel: i18n.BULK_EDIT_FLYOUT_FORM_ADD_INDEX_PATTERNS_LABEL,
+        indexHelpText: i18n.BULK_EDIT_FLYOUT_FORM_ADD_INDEX_PATTERNS_HELP_TEXT,
+        formTitle: i18n.BULK_EDIT_FLYOUT_FORM_ADD_INDEX_PATTERNS_TITLE,
+      }
+    : {
+        indexLabel: i18n.BULK_EDIT_FLYOUT_FORM_DELETE_INDEX_PATTERNS_LABEL,
+        indexHelpText: i18n.BULK_EDIT_FLYOUT_FORM_DELETE_INDEX_PATTERNS_HELP_TEXT,
+        formTitle: i18n.BULK_EDIT_FLYOUT_FORM_DELETE_INDEX_PATTERNS_TITLE,
+      };
+
+interface IndexPatternsFormProps {
+  editAction: IndexPatternsEditActions;
+  rulesCount: number;
+  onClose: () => void;
+  onConfirm: (bulkactionEditPayload: BulkActionEditPayload) => void;
+}
+
+const IndexPatternsFormComponent = ({
+  editAction,
+  rulesCount,
+  onClose,
+  onConfirm,
+}: IndexPatternsFormProps) => {
+  const { form } = useForm({
+    defaultValue: initialFormData,
+    schema,
+  });
+
+  const { indexHelpText, indexLabel, formTitle } = getFormConfig(editAction);
+
+  const [{ overwrite }] = useFormData({ form, watch: ['overwrite'] });
+  const { uiSettings } = useKibana().services;
+  const defaultPatterns = uiSettings.get<string[]>(DEFAULT_INDEX_KEY);
+
+  const handleSubmit = async () => {
+    const { data, isValid } = await form.submit();
+    if (!isValid) {
+      return;
+    }
+
+    const payload = {
+      value: data.index,
+      type: data.overwrite ? BulkActionEditType.set_index_patterns : editAction,
+    };
+
+    onConfirm(payload);
+  };
+
+  return (
+    <BulkEditFormWrapper form={form} onClose={onClose} onSubmit={handleSubmit} title={formTitle}>
+      <CommonUseField
+        path="index"
+        config={{ ...schema.index, label: indexLabel, helpText: indexHelpText }}
+        componentProps={{
+          idAria: 'detectionEngineBulkEditIndexPatterns',
+          'data-test-subj': 'detectionEngineBulkEditIndexPatterns',
+          euiFieldProps: {
+            fullWidth: true,
+            placeholder: '',
+            noSuggestions: false,
+            options: defaultPatterns.map((label) => ({ label })),
+          },
+        }}
+      />
+      {editAction === BulkActionEditType.add_index_patterns && (
+        <CommonUseField
+          path="overwrite"
+          componentProps={{
+            idAria: 'detectionEngineBulkEditOverwriteIndexPatterns',
+            'data-test-subj': 'detectionEngineBulkEditOverwriteIndexPatterns',
+          }}
+        />
+      )}
+      {overwrite && (
+        <EuiFormRow>
+          <EuiCallOut color="warning" size="s">
+            <FormattedMessage
+              id="xpack.securitySolution.detectionEngine.components.allRules.bulkActions.bulkEditFlyoutForm.setIndexPatternsWarningCallout"
+              defaultMessage="You’re about to overwrite index patterns for {rulesCount, plural, one {# selected rule} other {# selected rules}}, press Save to
+              apply changes."
+              values={{ rulesCount }}
+            />
```

Comment:
Nit: here we have similar difference in the number format:

![](https://puu.sh/IFyhp/c119f55ffc.png)